cmd/sqlcmd/main.go

```diff
@@ -201,16 +201,21 @@ func setConnect(connect *sqlcmd.ConnectSettings, args *SQLCmdArguments, vars *sq
 	connect.ErrorSeverityLevel = args.ErrorSeverityLevel
 }
 
+type allocConsole func(historyFile string) sqlcmd.Console
+
+var consoleAllocator allocConsole = console.NewConsole
+
 func run(vars *sqlcmd.Variables, args *SQLCmdArguments) (int, error) {
 	wd, err := os.Getwd()
 	if err != nil {
 		return 1, err
 	}
 
 	iactive := args.InputFile == nil && args.Query == ""
+	uactive := args.UserName != ""

```

Comment:
>.UserName != "" [](http://example.com/codeflow?start=16&length=15)

i think username can be non-empty and password be empty for some versions of AAD auth.